### PR TITLE
TESTING RPC

### PR DESCRIPTION
**Pull request by vimukthi-git**
Testrpc server + Enabling happy path tests

### CODE REVIEW

File: testrpc/server.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+func (s *stateService) GetStorage(key *string, blocknum *string) string {
+	if key != nil && blocknum != nil {
+		return s.storageForBlock[*key][*blocknum]
+	} else if key != nil {
+		return s.storage[*key]
+	}
+	return ""
+}
+
+type Server struct {
```
Comment (vimukthi-git) [inline testrpc/server.go]: main change

File: system/rpc_test.go [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
+
+func TestMain(m *testing.M) {
+	testServer = new(testrpc.Server)
+	var err error
+	rpcPort, err = testServer.Init(testrpc.GetTestMetaData())
+	if err != nil {
+		panic(err)
+	}
+
+	testClient, err = substrate.Connect("ws://localhost:" + strconv.Itoa(rpcPort))
```
Comment (pstehlik) [inline system/rpc_test.go]: would be good to allow passing in a different host

File: testrpc/test_metadata.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package testrpc
+
+func GetTestMetaData() string {
```
Comment (pstehlik) [inline testrpc/test_metadata.go]: does that mean we will have to seed the "Test data" with hex data each time?
or is it somehow possible to set this more "human friendly"?

Comment (vimukthi-git) [reply testrpc/test_metadata.go]: this metadata is for the reply of the metadata rpc call which has to be hex. But we can also encoded it on the fly from human friendly form. To keep it simple for now, I'm just using a response I got from our substrate chain. But we can change this in the future for sure.

**Pull request merged**